modules/uhttp/uhttpclient/client.go

```diff
@@ -24,15 +24,16 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/opentracing/opentracing-go"
 	"go.uber.org/fx/auth"
 )
 
 // New creates an http.Client that includes 2 extra outbound middleware: tracing and auth
 // they are going to be applied in following order: tracing, auth, remaining outbound middleware
 // and only if all of them passed the request is going to be send.
 // Client is safe to use by multiple go routines, if global tracer is not changed.
-func New(info auth.CreateAuthInfo, middleware ...OutboundMiddleware) *http.Client {
-	defaultMiddleware := []OutboundMiddleware{tracingOutbound(), authenticationOutbound(info)}
+func New(info auth.CreateAuthInfo, tracer opentracing.Tracer, middleware ...OutboundMiddleware) *http.Client {
```

Comment:
i think we should check for nils for info and tracer and set to nops, otherwise there may be crashes if called with `(nil, nil)` :(